src/factory.ts

```diff
@@ -0,0 +1,78 @@
+import { existsSync, lstatSync } from 'fs';
+import { stat } from 'node:fs/promises';
+import { extname } from 'path';
+import { FatFile } from './fat';
+import { MachoFile } from './macho';
+import filterAsync from 'node-filter-async';
+import {glob} from 'glob';
+
+const extensions = [
+    '.app',
+    '.dsym',
+    '.bundle',
+    '.framework',
+    '.xcarchive',
+    ''
+];
+
+export async function createMachoFiles(path: string): Promise<MachoFile[]> {
+    if (!path) {
+        throw new Error('Missing path to Mach-O file or Universal binary.');
+    }
+
+    if (!existsSync(path)) {
+        throw new Error(`File does not exist at ${path}`);
+    }
+
```

Comment:
Perhaps a comment here laying out what you are planning to do i.e.
* do this
* do that
* do this other thing
* then do this

That way the reader of this future code as a general idea what's coming and why?

Reply:
This is a reasonable ask. I don't think adding comments is the answer, but I could definitely write more self-documenting code!

Reply:
[34d4967](https://github.com/BugSplat-Git/macho-uuid/pull/1/commits/34d4967e66bca8d4f072d204e29c47a02d026fe1)

Reply:
I caved and created [9d91b1b](https://github.com/BugSplat-Git/macho-uuid/pull/1/commits/9d91b1b9bd7230c9bb602ae5eb34a804c849fb1d) as well... This stuff is pretty complicated 😭.